### PR TITLE
feat: add strategy management tests

### PR DESCRIPTION
**Pull request by jmonteer**
## Description

Add strategy management tests.

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

**Comment by jmonteer** [comment]
@anticlimactic can you handle these tests too?

**Comment by anticlimactic** [comment]
@jmonteer np will do

File: contracts/VaultV3.vy [hunk 1]
```diff
@@ -187,73 +200,80 @@ def withdraw(_shares: uint256, _recipient: address, _strategies: DynArray[addres
...
 @external
 def migrateStrategy(new_strategy: address, old_strategy: address):
-   # TODO: permissioned: STRATEGY_MANAGER
+    # TODO: permissioned: STRATEGY_MANAGER
 
-    # TODO: add strategy migrated event? 
-    assert self.strategies[old_strategy].activation != 0
-    assert self.strategies[old_strategy].currentDebt == 0
+    assert self.strategies[old_strategy].activation != 0, "old strategy not active"
+    assert self.strategies[old_strategy].currentDebt == 0, "old strategy has debt"
```
Comment (anticlimactic) [inline contracts/VaultV3.vy:272]: this is not currently being tested: we want to eventually allow migrations with strategies that have debt right, and this is just a placeholder?

**Comment by anticlimactic** [comment]
This PR is now ready for review @jmonteer @pandadefi 

I pulled the strategy mock classes from the other v3 repo and added a few missing tests. 

- For migrations, should I be testing if currentDebt == 0? I think we should support migrations with non-zero currentDebt (but maybe that can come later)
- I have also left out the two tests that require non-zero currentDebt (not implemented yet)

File: contracts/VaultV3.vy (hunk 1 above)
Comment (jmonteer) [reply contracts/VaultV3.vy:272]: you are right. this should be removed (I removed in other commit I think)

**Review by jmonteer — COMMENTED** [review]
LGTM
